lib/libutee/tee_api_objects.c

```diff
@@ -38,24 +38,80 @@
 #define TEE_ATTR_BIT_PROTECTED              (1 << 28)
 
 /* Data and Key Storage API  - Generic Object Functions */
+/*
+ * Use of this function is deprecated
+ * new code SHOULD use the TEE_GetObjectInfo1 function instead
+ * These functions will be removed at some future major revision of
+ * this specification
+ */
 void TEE_GetObjectInfo(TEE_ObjectHandle object, TEE_ObjectInfo *objectInfo)
 {
 	TEE_Result res;
 
 	res = utee_cryp_obj_get_info((uint32_t)object, objectInfo);
+
 	if (res != TEE_SUCCESS)
 		TEE_Panic(res);
+
+	slf_bridge_gp11_check_objectinfo_rc(objectInfo);
+}
+
+TEE_Result TEE_GetObjectInfo1(TEE_ObjectHandle object, TEE_ObjectInfo *objectInfo)
+{
+	TEE_Result res;
+
+	res = utee_cryp_obj_get_info((uint32_t)object, objectInfo);
+
+	if (res != TEE_SUCCESS && res != TEE_ERROR_CORRUPT_OBJECT &&
+	    res != TEE_ERROR_STORAGE_NOT_AVAILABLE)
+		TEE_Panic(res);
+
+	if (res == TEE_ERROR_CORRUPT_OBJECT) {
+		res = utee_storage_obj_del(object);
+		if (res != TEE_SUCCESS)
+			TEE_Panic(0);
+		return TEE_ERROR_CORRUPT_OBJECT;
+	}
+
+	return res;
 }
 
+/*
+ * Use of this function is deprecated
+ * new code SHOULD use the TEE_RestrictObjectUsage1 function instead
+ * These functions will be removed at some future major revision of
+ * this specification
+ */
 void TEE_RestrictObjectUsage(TEE_ObjectHandle object, uint32_t objectUsage)
 {
 	TEE_Result res;
-	res = utee_cryp_obj_restrict_usage((uint32_t)object, objectUsage);
+
+	res = TEE_RestrictObjectUsage1(object, objectUsage);
```

Comment:
This change is not 1.1 compliant:
*If the supplied object is persistent and corruption is detected then this function does nothing and returns. The object handle is not closed since the next use of the handle will return the corruption and delete it.*